### PR TITLE
meta: add type/assertion hints and upgrade pytest

### PR DESCRIPTION
**Pull request by cgsheeh**
- testing: add some assertion hints to test cases (Bug 1759890)
- typing: add type hints in a few places (Bug 1759890)
- testing: upgrade pytest to 7.1.1 (Bug 1754816)

### CODE REVIEW

**Comment by cgsheeh** [comment]
These are a few of the smaller changes in a larger patch stack - pulling them out and submitting them to make the stack smaller/review of the larger change easier.

File: landoapi/treestatus.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(self, *, url=None, session=None):
         self.url = self.url if self.url[-1] == "/" else self.url + "/"
         self.session = session or self.create_session()
 
-    def is_open(self, tree):
+    def is_open(self, tree) -> bool:
```
Comment (zzzeid) [inline landoapi/treestatus.py:32]: nit: Why not also add type hints for the arguments as well, here and below -- since you're here already?

**Review by zzzeid — APPROVED** [review]
Minor comment otherwise looks good.

File: landoapi/treestatus.py (hunk 1 above)
Comment (cgsheeh) [reply landoapi/treestatus.py:32]: Good point, I'll do that now.

**Pull request merged**